### DISCUSSION TITLE
having JSON output for "vpncmd" tools

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [SoftEtherVPN/SoftEtherVPN#1800] [Ideas]
For a few months working with both "vpncmd" and RPC-JSON API to develop a CLI to manage a SE server, unfortunately JSON API is buggy and incomplete and "vpncmd" tools do not have JSON output on in order to parse it. In summary 

 - RPC-JSON is buggy and incomplete
 - vpncmd is readable but not parsable

The idea is instead of developing two separate APIs (vpncmd + RPC) it could be just "vpncmd" tools that can output both:  
- normal text to read
- JSON format to parse for third-party tools/apps

like what "iproute2" tools do.

here is an example

*normal output*
```
ip route show 
default via 192.168.1.1 dev enp3s0 proto dhcp src 192.168.1.22 metric 100 
192.168.1.0/24 dev enp3s0 proto kernel scope link src 192.168.1.22 metric 100
```

*JSON output*
```
ip --json route show  | jq 
[
  {
    "dst": "default",
    "gateway": "192.168.1.1",
    "dev": "enp3s0",
    "protocol": "dhcp",
    "prefsrc": "192.168.1.22",
    "metric": 100,
    "flags": []
  },
  {
    "dst": "192.168.1.0/24",
    "dev": "enp3s0",
    "protocol": "kernel",
    "scope": "link",
    "prefsrc": "192.168.1.22",
    "metric": 100,
    "flags": []
  }
]
```

**Comment by shakibamoshiri** [comment]
I found `/CSV` option and tried, better than raw output, yet JSON could be much better

**Reply by metalefty** [reply]
agreed.
